serializer.go

```diff
@@ -264,6 +274,18 @@ func getIDorName(v map[string]any) (id string, name string) {
 	return
 }
 
+func getExistsCause(v map[string]any) (exists bool, cause string) {
+	if existsRaw, ok := v["exists"]; ok {
+		exists = existsRaw.(bool)
+		if causeRaw, hasCause := v["cause"]; hasCause {
+			cause = causeRaw.(string)
+			return exists, cause
+		}
```

Comment:
You aren't using `exists` unless `cause` is present, so you might as well wait to cast it until you've checked for cause.
```suggestion
		if causeRaw, hasCause := v["cause"]; hasCause {
		  return existsRaw.(bool), causeRaw.(string)
		}
```

Reply:
err, thanks I had intended to actually validate `exists == false` but I took the second half of your suggestion